server/v2/stf/stf.go

```diff
@@ -124,7 +125,7 @@ func (s STF[T]) DeliverBlock(
 		txResults[i] = s.deliverTx(ctx, newState, txBytes, corecontext.ExecModeFinalize, hi)
 	}
 	// end block
-	endBlockEvents, valset, err := s.endBlock(ctx, newState, hi)
+	endBlockEvents, valset, err := s.endBlock(exCtx)
```

Comment:
there is a mixed use of `ctx` and `exCtx` now. Would L125 benefit from `exCtx` as well?